### PR TITLE
Refactor CPU hotpath profiler and add option to filter out suspended threads

### PR DESCRIPTION
**Pull request by ogxd**

### CODE REVIEW

**Comment by ogxd** [comment]
Coucou @Bluezen, si tu as le temps de checker cette PR j'ai mis des informations dans l'issue qui est liée

File: src/DrDotnet.Profilers/profilers/cpu_hotpath_profiler.rs [hunk 1]
```diff
@@ -176,116 +41,175 @@ impl Profiler for CpuHotpathProfiler {
...
+        // Filter out unmanaged stack frames
+        if method_id == 0 {
+            return;
+        }
+        self.method_ids.push(method_id);
+        // Detect suspended threads appart from actual working threads
+        // Inspired from: https://www.usenix.org/legacy/publications/library/proceedings/coots99/full_papers/liang/liang_html/node10.html
+        // Not sure which is the best approach between utilizing the instruction pointer or the context
+        self.hasher.write_usize(instruction_pointer);
+        //self.hasher.write(context);
```
Comment (Bluezen) [inline src/DrDotnet.Profilers/profilers/cpu_hotpath_profiler.rs:78]: from my understanding context looks like the appropriate structure to hash (but I honestly don't really get instruction pointer). 
If we go for `instruction_pointer` we don't need the `COR_PRF_SNAPSHOT_CONTEXT` flag to `ICorProfilerInfo2::DoStackSnapshot` and `threads_by_context_hash` can be renamed `threads_by_instruction_pointer_hash`

Comment (ogxd) [reply src/DrDotnet.Profilers/profilers/cpu_hotpath_profiler.rs]: I've ran a test in production using instruction pointer, I'll run another test using the context and see how it differs

Comment (ogxd) [reply src/DrDotnet.Profilers/profilers/cpu_hotpath_profiler.rs]: Both seem to yield comparable results on production

Comment (ogxd) [reply src/DrDotnet.Profilers/profilers/cpu_hotpath_profiler.rs]: I'll stick with instruction_pointer which is probably more lightweight

**Review by Bluezen — APPROVED** [review]

**Pull request merged**